### ISSUE TITLE
UUID should support to and from standard uuid formats.

### ISSUE THREAD

**Issue by kelvinhammond** [openzim/libzim#680]
The UUID format should support converting to and from the standard display format(s).
See https://en.wikipedia.org/wiki/Universally_unique_identifier#Format

https://github.com/openzim/libzim/blob/master/src/uuid.cpp

**Comment by kelson42** [comment]
I agree with this request, this one I made more than 10 years ago and I still think this would be handy... at least the one to get a printable string.

**Comment by mgautierfr** [comment]
Despite its name, zim's uuid is only a unique id stored in 128 bits, it is not a standard UUID  (which is also a unique id stored in 128 bits, but with more constraints)
None of the zim's uuid generated by libzim are valid standard UUID.

But anyway, there is already a (two) way(s) to convert a uuid to "standard" formats 8-4-4-4-12 (https://github.com/openzim/libzim/blob/master/src/uuid.cpp#L86-L110)
The operator `<<` to print a UUID into a stream and the `string` operator to convert a UUID to a string.

UUID can also be constructed from raw binary data (which is not a standard representation). We could indeed add a constructor from a textual representation 8-4-4-4-12 to help manipulation of the uuid.

**Comment by kelson42** [comment]
@kelvinhammond Good enough?

**Comment by kelvinhammond** [comment]
I'll bind this to nodejs as is and whatever library wants to use `setUuid` will have to figure out which chars to send. This should suffice.

**Comment by kelson42** [comment]
@mgautierfr Would that be a good id that in the future we allow only proper uuids?

**Comment by mgautierfr** [comment]
I don't see the added value to this. We would need to depend on a uuid library to generate the uuid (or check it is a valid one) and it would be useless (old zim files would still have invalid uuid so we couldn't convert zim uuid to standard UUID)

**Comment by kelson42** [comment]
@mgautierfr The `README.md` states tha we already depends of libuuid? Is that true? I don’t see any trace of a check in the Meson configuration file!

**Comment by mgautierfr** [comment]
It is false. We don't use libuuid.

**Comment by kelson42** [comment]
@kelvinhammond It seems we have a way in the libzim to print the UUID. Does the answer of @mgautierfr is satisfying to you?
